naxsi_src/naxsi_utils.c

```diff
@@ -973,7 +971,7 @@ naxsi_log_offending_as_json(ngx_http_request_ctx_t* ctx,
   }
 
   // request id
-  out    = naxsi_log_as_json_string(out, end, "rid", req_id, NAXSI_REQUEST_ID_STRLEN - 1);
+  out    = naxsi_log_as_json_string(out, end, "rid", req_id, strlen((const char*)req_id));
```

Comment:
there is a NULL deref here. if req_id is null for any reason, this will crash badly.